lib/mix/lib/mix/local/installer.ex

```diff
@@ -241,26 +241,45 @@ defmodule Mix.Local.Installer do
     {opts, argv} = OptionParser.parse!(argv, switches: switches)
 
     if name = List.first(argv) do
-      path = Path.join(root, name)
+      found =
+        cond do
+          File.exists?(Path.join(root, name)) -> Path.join(root, name)
+          matching = matching_package(root, name) -> Path.join(root, matching)
+          true -> nil
+        end
 
       cond do
-        not File.exists?(path) ->
-          Mix.shell().error("Could not find a local artifact named #{inspect(name)}. We found:")
-          Mix.Task.rerun(listing)
-          nil
+        found && should_uninstall?(found, opts) ->
+          File.rm_rf!(found)
+          found
 
-        should_uninstall?(path, opts) ->
-          File.rm_rf!(path)
-          path
+        found ->
+          nil
 
         true ->
+          Mix.shell().error("Could not find a local artifact named #{inspect(name)}. We found:")
+          Mix.Task.rerun(listing)
           nil
       end
     else
       Mix.raise("No argument was given to uninstall command")
     end
   end
 
+  defp matching_package(root, name) do
```

Comment:
I think you can use Path.wildcard here. Then you can rewrite this function to: `Path.wildcard(Path.join(root, name <> "-*") |> List.first()`

Reply:
👍 done